Core/ActsVersion.hpp.in

```diff
@@ -26,4 +26,23 @@ constexpr unsigned int Version =
 constexpr const char* const CommitHash = "@_acts_commit_hash@";
 constexpr const char* const CommitHashShort = "@_acts_commit_hash_short@";
 
+struct VersionInfo {
+  unsigned int major;
+  unsigned int minor;
+  unsigned int patch;
+  const char* const commitHash;
+  const char* const commitHashShort;
+
+  VersionInfo();
+
+  static bool isConsistent() {
```

Comment:
can this be `constexpr` or be implemented in the source file?

Reply:
The idea is that it is implement in the header. This way I can check if the shared lib and the header that some compilation unit was looking at report the same version.